File: ParadigmaComp/15Questoes/golang/questao2.go
package main

import "fmt"
// Métodos para Acelerar e Frear

type Carro struct {
	Marca     string
	Modelo    string
	Ano       int
	Velocidade int
}

func (c *Carro) Acelerar(valor int) {
	c.Velocidade += valor
}

func (c *Carro) Frear(valor int) {
	c.Velocidade -= valor
	if c.Velocidade < 0 {
		c.Velocidade = 0
	}
}

func (c Carro) ExibirVelocidade() string {
	return fmt.Sprintf("Velocidade atual: %d km/h", c.Velocidade)
}

func main() {
	carro := Carro{"Toyota", "Corolla", 2020, 0}
	carro.Acelerar(50)
	fmt.Println(carro.ExibirVelocidade())
	carro.Frear(20)
	fmt.Println(carro.ExibirVelocidade())
}

File: ParadigmaComp/15Questoes/python/questao2.py
# Métodos para Acelerar e Frear
class Carro:
    def __init__(self, marca, modelo, ano):
        self.marca = marca
        self.modelo = modelo
        self.ano = ano
        self.velocidade = 0

    def acelerar(self, valor):
        self.velocidade += valor

    def frear(self, valor):
        self.velocidade -= valor
        if self.velocidade < 0:
            self.velocidade = 0

    def exibir_velocidade(self):
        return f"Velocidade atual: {self.velocidade} km/h"

# Exemplo de uso
carro = Carro("Toyota", "Corolla", 2020)
carro.acelerar(50)
print(carro.exibir_velocidade())
carro.frear(20)
print(carro.exibir_velocidade())
